src/main/java/dev/sigstore/KeylessVerifier.java

```diff
@@ -0,0 +1,197 @@
+      this.fulcioVerifier = fulcioVerifier;
+      return this;
+    }
+
+    public KeylessVerifier.Builder rekorClient(
+        RekorClient rekorClient, RekorVerifier rekorVerifier) {
+      this.rekorClient = rekorClient;
+      this.rekorVerifier = rekorVerifier;
+      return this;
+    }
+
+    public KeylessVerifier build() {
+      Preconditions.checkNotNull(fulcioVerifier);
+      Preconditions.checkNotNull(rekorVerifier);
+      Preconditions.checkNotNull(rekorClient);
+      return new KeylessVerifier(fulcioVerifier, rekorClient, rekorVerifier);
+    }
+
+    public Builder sigstorePublicDefaults()
+        throws IOException, InvalidAlgorithmParameterException, CertificateException,
+            InvalidKeySpecException, NoSuchAlgorithmException {
+      var fulcioCert =
+          Resources.toByteArray(
+              Resources.getResource("dev/sigstore/tuf/production/fulcio_v1.crt.pem"));
+      var ctfePublicKey =
+          Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/production/ctfe.pub"));
+      var rekorPublicKey =
+          Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/production/rekor.pub"));
+      fulcioVerifier(FulcioVerifier.newFulcioVerifier(fulcioCert, ctfePublicKey));
+      rekorClient(RekorClient.builder().build(), RekorVerifier.newRekorVerifier(rekorPublicKey));
+      return this;
+    }
+
+    public Builder sigstoreStagingDefaults()
+        throws IOException, InvalidAlgorithmParameterException, CertificateException,
+            InvalidKeySpecException, NoSuchAlgorithmException {
+      var fulcioCert =
+          Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/staging/fulcio.crt.pem"));
+      var ctfePublicKey =
+          Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/staging/ctfe.pub"));
+      var rekorPublicKey =
+          Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/staging/rekor.pub"));
+      fulcioVerifier(FulcioVerifier.newFulcioVerifier(fulcioCert, ctfePublicKey));
+      rekorClient(
+          RekorClient.builder().setServerUrl(URI.create(RekorClient.STAGING_REKOR_SERVER)).build(),
+          RekorVerifier.newRekorVerifier(rekorPublicKey));
+      return this;
+    }
+  }
+
+  /**
+   * Verify that the inputs can attest to the validity of a signature using sigstore's keyless
+   * infrastructure. If no exception is thrown, it should be assumed verification has passed.
+   *
+   * @param artifactDigest the sha256 digest of the artifact that was signed
+   * @param certChain the certificate chain obtained from a fulcio instance
+   * @param signature the signature on the artifact
+   * @throws KeylessVerificationException if the signing information could not be verified
+   * @throws IOException if an error occurred communicating with remote rekor
+   */
+  public void verifyOnline(byte[] artifactDigest, byte[] certChain, byte[] signature)
```

Comment:
Most of the formats I've seen so store cert and cert chain separately. https://docs.google.com/document/d/1gucjOA_bGyRjK6TeaOI-X5GIUv8WsPzeMDMkq25Kv4Y/edit is going to get implemented and also does so. Should we be accepting them as separate params?

Reply:
hrmm... I dunno, we'd have to combine them together anyway during verification. I think we can accept them as separate params in the future as we work things out. But I think this is sufficient for now? I could be convinced otherwise.